app/src/full/java/io/homeassistant/companion/android/settings/wear/views/SettingsWearTemplateTileList.kt

```diff
@@ -0,0 +1,88 @@
+package io.homeassistant.companion.android.settings.wear.views
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.mikepenz.iconics.typeface.library.community.material.CommunityMaterial
+import io.homeassistant.companion.android.common.data.prefs.impl.entities.TemplateTileConfig
+import io.homeassistant.companion.android.settings.views.SettingsRow
+import io.homeassistant.companion.android.common.R as commonR
+
+@Composable
+fun SettingsWearTemplateTileList(
+    templateTiles: Map<Int, TemplateTileConfig>,
+    onTemplateTileClicked: (tileId: Int) -> Unit,
+    onBackClicked: () -> Unit
+) {
+    Scaffold(
+        topBar = {
+            SettingsWearTopAppBar(
+                title = { Text(stringResource(commonR.string.template_tiles)) },
+                onBackClicked = onBackClicked,
+                docsLink = WEAR_DOCS_LINK
+            )
+        }
+    ) { padding ->
+        Column(
+            Modifier
+                .padding(padding)
+                .padding(all = 16.dp)
+        ) {
+            if (templateTiles.entries.isEmpty()) {
+                Text(stringResource(commonR.string.template_tile_no_tiles_yet))
+            } else {
+                var index = 1
+                for (templateTileEntry in templateTiles.entries) {
+                    SettingsRow(
+                        primaryText = stringResource(commonR.string.template_tile_n, index++),
+                        secondaryText = stringResource(commonR.string.template_tile_configure),
```

Comment:
If you have more than one tile, this is a bit repetitive for every row. Suggestion: add it as a header to the list (similar to the 'Manage [Wear device] header' in the main screen).

Reply:
Good idea - and fully agreed about the repetition :+1: Added it as a header.

`SettingsRow` requires a `secondaryText` to be provided though (as far as I understand it), and by giving it an empty string, the alignment looks a bit off. Any suggestions?

Making `secondaryText` optional and only outputting it if it's non-null seems to fix the alignment at least:

```diff
diff --git app/src/main/java/io/homeassistant/companion/android/settings/views/SettingsRow.kt app/src/main/java/io/homeassistant/companion/android/settings/views/SettingsRow.kt
index 01231d2a..190884e0 100644
--- app/src/main/java/io/homeassistant/companion/android/settings/views/SettingsRow.kt
+++ app/src/main/java/io/homeassistant/companion/android/settings/views/SettingsRow.kt
@@ -28,7 +28,7 @@ import com.mikepenz.iconics.typeface.IIcon
 @Composable
 fun SettingsRow(
     primaryText: String,
-    secondaryText: String,
+    secondaryText: String?,
     mdiIcon: IIcon?,
     enabled: Boolean,
     onClicked: () -> Unit
@@ -66,11 +66,13 @@ fun SettingsRow(
                 text = primaryText,
                 style = MaterialTheme.typography.body1
             )
-            CompositionLocalProvider(LocalContentAlpha provides ContentAlpha.medium) {
-                Text(
-                    text = secondaryText,
-                    style = MaterialTheme.typography.body2
-                )
+            secondaryText?.let {
+                CompositionLocalProvider(LocalContentAlpha provides ContentAlpha.medium) {
+                    Text(
+                        text = it,
+                        style = MaterialTheme.typography.body2
+                    )
+                }
             }
         }
     }
```

Reply:
If you need to show something as the secondary label, maybe show the template (or the first 100 characters to prevent a very long template from filling the screen)? The current fix with an optional secondary label is also fine.

Reply:
Hmm, that's a really good idea. Only the numbers make it hard to know which tile a specific one is. Showing the template would make it a little bit easier :ok_hand: Fixed like that instead.